File: bin/materi3/Queue.py
class Queue:
    def __init__(self, size):
        self.maxSize = size
        self.quaArray = [None] * size
        self.front = 0
        self.rear = -1
        self.nItems = 0

    def insert(self, value):
        if self.rear == self.maxSize - 1:
            self.rear = -1
        self.rear += 1
        self.quaArray[self.rear] = value
        self.nItems += 1

    def remove(self):
        temp = self.quaArray[self.front]
        self.front += 1
        if self.front == self.maxSize:
            self.front = 0
        self.nItems -= 1
        return temp

    def peekFront(self):
        return self.quaArray[self.front]

    def isEmpty(self):
        return self.nItems == 0

    def isFull(self):
        return self.nItems == self.maxSize

    def size(self):
        return self.nItems

if __name__ == "__main__":
    theQueue = Queue(100)

    data = input("Input keyboard: ")

    for c in data:
        theQueue.insert(c)

    while not theQueue.isEmpty():
        temp = theQueue.remove()
        print(temp, end=", ")

File: src/materi3/Queue.java
package materi3;

public class Queue {
    private int maxSize;
    private char[] quaArray;
    private int front;
    private int rear;
    private int nItems;

    public Queue(int size) {
        this.maxSize = size;
        quaArray = new char[maxSize];
        front = 0;
        rear = -1;
        nItems = 0;
    }

    public void insert(char value) {
        if (rear == maxSize - 1) {
            rear = -1;
        }
        quaArray[++rear] = value;
        nItems++;
    }

    public char remove() {
        char temp = quaArray[front++];
        if (front == maxSize) {
            front = 0;
        }
        nItems--;
        return temp;
    }

    public char peekFront() {
        return quaArray[front];
    }

    public boolean isEmpty() {
        return (nItems == 0);
    }

    public boolean isFull() {
        return (nItems == maxSize);
    }

    public int size() {
        return nItems;
    }
}
